typing/types.ml

```diff
@@ -300,6 +300,16 @@ and constructor_tag =
   | Cstr_extension of Path.t * bool     (* Extension constructor
                                            true if a constant false if a block*)
 
+let compare_tag t1 t2 = match (t1, t2) with
+  | Cstr_constant i1, Cstr_constant i2 -> i2 - i1
+  | Cstr_block i1, Cstr_block i2 -> i2 - i1
+  | Cstr_extension (path1, b1), Cstr_extension (path2, b2) -> (match compare path1 path2 with
+    | 0 -> compare b1 b2
+    | c -> c)
+  | Cstr_constant _, _ -> 1
+  | Cstr_block _, _ -> 1
+  | _ -> -1
```

Comment:
That seems wrong. `compare_tag (Cstr_constant _) (Cst_block _) = compare_tag (Cst_block _)  (Cstr_constant _) = 1`
Also, while using substraction as comparison is ok for small integers, it's not really faster than the correct version anyway.

You only need `equal_tag` anyway.